### PR TITLE
fix: innerHTML does not inject variables

### LINKED ISSUE #206: innerHTML broken for some frameworks
[Reproduction](
https://mitosis.builder.io/?outputTab=react&code=export+default+function+MyComponent%28props%29+%7B%0A++return+%28%0A++++%3Cdiv+innerHTML%3D%7B%27%3Cp+class%3D%22text-lg%22%3E%3C%2Fp%3E%27%7D%3E%0A++++++Hello%21+I+can+run+in+React%2C+Vue%2C+Solid%2C+or+Liquid%21%0A++++%3C%2Fdiv%3E%0A++%29%3B%0A%7D)

For React, React Native, Vue, and Angular there are two sets of quotes.

For Svelte it should use [`@html`](https://svelte.dev/tutorial/html-tags).

### LINKED ISSUE #286: React: innerHTML does not inject variables
**Describe the bug**
`innerHTML={state.name}` is stringified as the HTML string "name" instead of referring to the variable:

```
<div className="div" dangerouslySetInnerHTML={{ __html: "name" }} />
```

**To Reproduce**
If possible, a link to a https://mitosis.builder.io/ fiddle containing the bug: [link](https://mitosis.builder.io/?outputTab=react&code=import+%7B+useState+%7D+from+%27%40builder.io%2Fmitosis%27%3B%0A%0Aexport+default+function+MyComponent%28props%29+%7B%0A++const+state+%3D+useState%28%7Bname%3A%27Steve%27%7D%29%3B%0A%0A++return+%28%0A++++%3Cdiv%0A++++++css%3D%7B%7B%0A++++++++color%3A+%27red%27%2C%0A++++++%7D%7D%0A++++++innerHTML%3D%7Bstate.name%7D%0A++++%3E%3C%2Fdiv%3E%0A++%29%3B%0A%7D%0A)

**Expected behavior**
```
<div className="div" dangerouslySetInnerHTML={{ __html: name }} />
```

### PR DESCRIPTION
**Pull request by decadef20**
## Description

Add a short description of:
- what changes you made, 
- why you made them, and 
- any other context that you think might be helpful for someone to better understand what is contained in this pull request. 

   [fix #286 ](https://github.com/BuilderIO/mitosis/issues/286)
   [fix #206 ](https://github.com/BuilderIO/mitosis/issues/206)

This sort of information is useful for people reviewing the code, as well as anyone from the future trying to understand why changes were made or why a bug started happening.

### CODE REVIEW

**Comment by decadef20** [comment]
fix #206 #286

File: packages/core/src/__tests__/__snapshots__/vue.test.ts.snap [hunk 1]
```diff
@@ -963,21 +963,37 @@ export default {
...
+
+
+    export default {
+      name: 'text',
+      
+      props: [\\"text\\",\\"content\\"],
+      
+      data: () => ({ name: 'Decadef20',}),
+      
+
```
Comment (PatrickJS) [inline packages/core/src/__tests__/__snapshots__/vue.test.ts.snap:984]: something is not right with this snap since the format is weird

Comment (PatrickJS) [reply packages/core/src/__tests__/__snapshots__/vue.test.ts.snap:984]: do you think its the `,` in 

`data: () => ({ name: 'Decadef20',}),`

`'Decadef20',}` the `,`

Comment (PatrickJS) [reply packages/core/src/__tests__/__snapshots__/vue.test.ts.snap:984]: can you look into this problem another PR

**Review by PatrickJS — APPROVED** [review]

File: packages/core/src/__tests__/__snapshots__/vue.test.ts.snap (hunk 1 above)
Comment (decadef20) [reply packages/core/src/__tests__/__snapshots__/vue.test.ts.snap:984]: OK, I'll fix it later

**Pull request merged**